docs/proposals/multiple-sources-for-applications.md

```diff
@@ -0,0 +1,225 @@
+) to expose Prometheus metrics.
+https://github.com/argoproj/argo-cd/issues/677
+
+#### Use case 2: 
+We have a Helm Chart which is used in 30+ Services and each of them is customized for 3 possible environments.
+Replicating this Chart 30 times without a centralized Repo looks dirty. Can be a show stopper for the whole migration.
+Modifying the Applica`tion definition is not an option since the whole goal is to reduce the rights that the CI-solution has. Giving it the right to update all Application-definitions from various teams in the argocd namespace is a a hard thing to convince people with.
+
+
+### Implementation Details
+
+#### Attach multiple sources to the Application
+
+To allow multiple sources to the Application, we would add a new field `sources` which would allow users to input list of `ApplicationSource`s. As part of this update and to support backward compatibilty, we would mark field `source` as deprecated and remove it in later revisions.
+
+To avoid complexity on the deciding the list of sources, if both `source` and `sources` fields are specified, we would override the source under `source` field with all the sources under `sources` field.
+
+#### Invalidating existing cache
+
+Argo CD benefits from the assumption of a single repo per application to detect changes and to cache the results. But this enhancement now requires us to look at all the source repo "HEAD"s and invalidate the cache if any one of them changes.
+
+#### Reconcilation of the Application
+
+As we would have multiple sources as part of the same Application, we would need to track updates to each source and reconcile the Application for each source. When one of the sources change, we would need to ensure that target revisions of other sources are up-to-date, e.g. force a simple refresh to see if target revision of the source (e.g. HEAD), still points to revisionX for example.
+
+#### Updates to UI
+Today, we allow users to add only one source in the UI. We would need to update the UI to add multiple sources and configure specific 
+
+#### Updates to cli
+
+We would need to create new options to the `argocd app create` command for adding multiple sources to the Application.
+
+For supporting external Helm value files, we would need to introduce a new option similar to existing `--values` option for Helm projects to support external values files. This options would need to be added individually to each source.
+
+
+### Security Considerations
+
+Good unit- and end-to-end tests need to be in place for this functionality to ensure we don't accidentally introduce a change that would allow any form of uncontrolled association between an Application and its resources.
```

Comment:
> uncontrolled association between an Application and its resources

I'm not sure I follow this. Could you expand on what that uncontrolled association might look like?